node/src/accountant/scanners/scan_schedulers.rs

```diff
@@ -152,49 +150,51 @@ impl PayableScanScheduler {
     }
 }
 
-pub struct PayableScanSchedulerInner {
-    pub last_new_payable_scan_timestamp: SystemTime,
-}
+pub trait NewPayableScanDynIntervalComputer {
+    fn compute_interval(&self) -> Option<Duration>;
 
-impl Default for PayableScanSchedulerInner {
-    fn default() -> Self {
-        Self {
-            last_new_payable_scan_timestamp: UNIX_EPOCH,
-        }
-    }
-}
+    fn zero_out(&mut self);
 
-pub trait NewPayableScanDynIntervalComputer {
-    fn compute_interval(
-        &self,
-        now: SystemTime,
-        last_new_payable_scan_timestamp: SystemTime,
-        interval: Duration,
-    ) -> Option<Duration>;
+    as_any_ref_in_trait!();
 }
 
-#[derive(Default)]
-pub struct NewPayableScanDynIntervalComputerReal {}
+pub struct NewPayableScanDynIntervalComputerReal {
+    scan_interval: Duration,
+    last_scan_timestamp: SystemTime,
+    clock: Box<dyn SimpleClock>,
+}
 
 impl NewPayableScanDynIntervalComputer for NewPayableScanDynIntervalComputerReal {
-    fn compute_interval(
-        &self,
-        now: SystemTime,
-        last_new_payable_scan_timestamp: SystemTime,
-        interval: Duration,
-    ) -> Option<Duration> {
+    fn compute_interval(&self) -> Option<Duration> {
```

Comment:
These names are too vague, they don't represent what you're trying to achieve here.

Please use this instead:

```rust
fn compute_scan_timing(&self) -> ScanTiming {
    let current_time = self.clock.now();
    let time_since_last_scan = current_time
        .duration_since(self.last_scan_timestamp)
        .unwrap_or_else(|_| {
            panic!(
                "Current time ({:?}) is earlier than last scan timestamp ({:?})",
                current_time, self.last_scan_timestamp
            )
        });
    
    if time_since_last_scan >= self.scan_interval {
        ScanTiming::ReadyNow
    } else {
        ScanTiming::WaitFor(self.scan_interval - time_since_last_scan)
    }
}
```

Reply:
Huh. Fairly annoyed, I did do it. 

Sometimes you're overdoing as a reviewer. This was on the edge. Note for the next time that Option may be much a better choice, as it has the whole pallet of methods and so you can easily work with it. Not so with a newly constructed enum. Still, I understand your intent. And as said, I did agree to go through the hassle. Just please don't push too much in refactoring.